internal/models/base/base_analyzer_test.go

```diff
@@ -0,0 +1,39 @@
+package base
+
+import (
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+type BaseAnalyzerTestSuite struct {
+	suite.Suite
+	baseAnalyzer *BaseAnalyzer
+}
+
+func TestBaseAnalyzerTestSuite(t *testing.T) {
+	suite.Run(t, new(BaseAnalyzerTestSuite))
+}
+
+func (s *BaseAnalyzerTestSuite) SetupSuite() {
+	s.baseAnalyzer = New()
+}
+
+func (s *BaseAnalyzerTestSuite) TestCompareVersionReturnsZero() {
```

Comment:
IMO, testing the logic is more appreciated. Plus that each test case should exactly test one concern.  
I prefer tests like:
- TestCanCompareMajorVersion
- TestCanCompareMinorVersion
- TestCanComparePatchVersion
- TestCanCompareVersionsWithPrefixV
- TestZeroIsReturnedForEqualVersions